jdaviz/configs/imviz/tests/test_parser.py

```diff
@@ -88,6 +88,23 @@ def test_parse_numpy_array(self, imviz_app):
         assert data.shape == (2, 2)
         assert comp.data.shape == (2, 2)
 
+    def test_parse_numpy_array_3d(self, imviz_app):
+        # This data has values in axis=0 that correspond to axis num.
+        n_slices = 5
+        slice_shape = (2, 3)
+        arr = np.stack([np.zeros(slice_shape) + i for i in range(n_slices)])
+
+        # We use higher level load_data() here to make sure linking does not crash.
+        imviz_app.load_data(arr, data_label='my_slices')
+        assert len(imviz_app.app.data_collection) == n_slices
```

Comment:
is it worth adding testing of the for-loop deferred linking use case?

Reply:
Yes, good call. I'll add that!

Reply:
Test added.